cloud/azure/azure_rm_deployment.py

```diff
@@ -0,0 +1,644 @@
+        "instances": [
+            {
+                "ips": [
+                    {
+                        "dns_settings": {
+                            "domain_name_label": "testvm9910001",
+                            "fqdn": "testvm9910001.westus.cloudapp.azure.com"
+                        },
+                        "id": "/subscriptions/3f7e29ba-24e0-42f6-8d9c-5149a14bda37/resourceGroups/Test_Deployment/providers/Microsoft.Network/publicIPAddresses/myPublicIP",
+                        "name": "myPublicIP",
+                        "public_ip": "13.91.99.232",
+                        "public_ip_allocation_method": "IPAllocationMethod.dynamic"
+                    }
+                ],
+                "vm_name": "MyUbuntuVM"
+            }
+        ],
+        "name": "ansible-arm",
+        "outputs": {
+            "hostname": {
+                "type": "String",
+                "value": "testvm9910001.westus.cloudapp.azure.com"
+            },
+            "sshCommand": {
+                "type": "String",
+                "value": "ssh chouseknecht@testvm9910001.westus.cloudapp.azure.com"
+            }
+        }
+    }
+'''
+
+import time
+import yaml
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.azure_rm_common import *
+
+try:
+    from itertools import chain
+    from azure.common.credentials import ServicePrincipalCredentials
+    from azure.common.exceptions import CloudError
+    from azure.mgmt.resource.resources.models import (DeploymentProperties,
+                                                      ParametersLink,
+                                                      TemplateLink,
+                                                      Deployment,
+                                                      ResourceGroup,
+                                                      Dependency)
+    from azure.mgmt.resource.resources import ResourceManagementClient, ResourceManagementClientConfiguration
+    from azure.mgmt.network import NetworkManagementClient, NetworkManagementClientConfiguration
+
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMDeploymentManager(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group_name=dict(type='str', required=True, aliases=['resource_group']),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            template=dict(type='dict', default=None),
+            parameters=dict(type='dict', default=None),
+            template_link=dict(type='str', default=None),
+            parameters_link=dict(type='str', default=None),
+            location=dict(type='str', default="West US"),
+            deployment_mode=dict(type='str', default='Complete', choices=['Complete', 'Incremental']),
```

Comment:
IIRC the Azure SDK is case-insensitive- if so, these should probably be lowercase (since they won't care, and our args are usually lowercased)